python/tvm/autotvm/graph_tuner/utils/infer_layout_transform.py

```diff
@@ -0,0 +1,81 @@
+# pylint: disable=too-many-locals
+"""Built-in functions to infer layout transformation."""
+
+def infer_conv2d_layout_shape_avx(wkl, current_sch, target_sch, multi_input_node_shape=None):
```

Comment:
can we leverage the c++ infer_shape implement of `layout_transform` instead of writing another py function? again, if it binds to an operator (e.g., conv2d), we'd better make the operator self-declare the behavior, like the registered FCompute/FInferShape/etc.

Reply:
Sure. It's ideal to implement this as another "attribute" method of operator. Let me consider a more systematic patten which can deal with layout transform interactions between different op, such as conv2d and depthwise-conv2d.